src/openrct2-ui/windows/Footpath.cpp

```diff
@@ -1034,7 +1034,7 @@ class FootpathWindow final : public Window
 
         auto footpathPlaceAction = FootpathPlaceAction(
             footpathLoc, slope, type, gFootpathSelection.Railings, _footpathConstructDirection, constructFlags);
-        footpathPlaceAction.SetCallback([=](const GameAction* ga, const GameActions::Result* result) {
+        footpathPlaceAction.SetCallback([this, footpathLoc](const GameAction* ga, const GameActions::Result* result) {
```

Comment:
ooof, We should never capture the window pointer in a callback. The window might have been closed and deleted by the time the callback is run. This needs to be reworked.

Reply:
It took me by surprise as well. Either way, this PR only makes explicit what was implicit before. I hope we can defer reworking this to a future PR.

Reply:
It doesn't seem to used it in the lambda, that would have lead to crashes for sure.

Reply:
Hmm. The only reason I'm capturing `this` explicitly here is because I was getting a compiler error warning me about implicitly capturing it being deprecated.

I'll happily revert, or indeed try only capturing `footpathLoc` explicitly?

Reply:
Without, I get the following errors:
```
[ 89%] Building CXX object CMakeFiles/openrct2.dir/src/openrct2-ui/windows/Footpath.cpp.o
/Users/aaron/Projects/OpenRCT2/src/openrct2-ui/windows/Footpath.cpp:1044:21: error: 'this' cannot be implicitly captured in this context
                    _footpathConstructValidDirections = INVALID_DIRECTION;
                    ^
/Users/aaron/Projects/OpenRCT2/src/openrct2-ui/windows/Footpath.cpp:1037:53: note: explicitly capture 'this'
        footpathPlaceAction.SetCallback([footpathLoc](const GameAction* ga, const GameActions::Result* result) {
                                                    ^
                                                    , this
/Users/aaron/Projects/OpenRCT2/src/openrct2-ui/windows/Footpath.cpp:1048:21: error: 'this' cannot be implicitly captured in this context
                    _footpathConstructValidDirections = _footpathConstructDirection;
                    ^
/Users/aaron/Projects/OpenRCT2/src/openrct2-ui/windows/Footpath.cpp:1037:53: note: explicitly capture 'this'
        footpathPlaceAction.SetCallback([footpathLoc](const GameAction* ga, const GameActions::Result* result) {
                                                    ^
                                                    , this
/Users/aaron/Projects/OpenRCT2/src/openrct2-ui/windows/Footpath.cpp:1048:57: error: 'this' cannot be implicitly captured in this context
                    _footpathConstructValidDirections = _footpathConstructDirection;
                                                        ^
/Users/aaron/Projects/OpenRCT2/src/openrct2-ui/windows/Footpath.cpp:1037:53: note: explicitly capture 'this'
        footpathPlaceAction.SetCallback([footpathLoc](const GameAction* ga, const GameActions::Result* result) {
                                                    ^
                                                    , this
/Users/aaron/Projects/OpenRCT2/src/openrct2-ui/windows/Footpath.cpp:1065:13: error: 'this' cannot be implicitly captured in this context
            WindowFootpathSetEnabledAndPressedWidgets();
            ^
/Users/aaron/Projects/OpenRCT2/src/openrct2-ui/windows/Footpath.cpp:1037:53: note: explicitly capture 'this'
        footpathPlaceAction.SetCallback([footpathLoc](const GameAction* ga, const GameActions::Result* result) {
                                                    ^
                                                    , this
4 errors generated.
```

Reply:
I would just reget the window as the first item in the game command and if it is valid then do whatever.